### PR TITLE
feat: make `--random-salt` default to forc-deploy and remove it as a flag

### LINKED ISSUE #4758: forc-deploy should have `--random-salt` by default
forc-deploy needs to have `--random-salt` enabled by default. If users specified a salt then we should use the user specified salt but in other cases we should automatically apply `--random-salt`.

### PR DESCRIPTION
**Pull request by kayagokalp**
## Description
closes #4758.

This PR makes --random-salt the default behaviour of forc-deploy, if salt information is passed, the passed salt would be used but otherwise, random-salt would be implied by default. Also adds --default-salt for CI so that in CI we can always create the same contract id if we want to. Useful for reproducible deployments.

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by kayagokalp** [comment]
Waiting on #4752.

File: forc-plugins/forc-client/src/cmd/deploy.rs [hunk 1]
```diff
@@ -29,10 +29,10 @@ pub struct Command {
     /// --salt contract_b:0x0000000000000000000000000000000000000000000000000000000000000002
     #[clap(long)]
     pub salt: Option<Vec<String>>,
-    /// Generate a random salt for the contract.
-    /// Useful for testing or deploying examples to a shared network.
+    /// Generate a defualt salt (0x0000000000000000000000000000000000000000000000000000000000000000) for the contract.
```
Comment (JoshuaBatty) [inline forc-plugins/forc-client/src/cmd/deploy.rs:32]:
```suggestion
    /// Generate a default salt (0x0000000000000000000000000000000000000000000000000000000000000000) for the contract.
```

**Review by JoshuaBatty — COMMENTED** [review]
Just one spelling mistake otherwise happy to approve

**Review by JoshuaBatty — APPROVED** [review]
utACK

**Review by eureka-cpu — APPROVED** [review]

**Pull request merged**